core/tee/tee_fs_common.c

```diff
@@ -37,19 +40,939 @@
+	struct tee_fs_rpc head = { 0 };
+
+	if (!len) {
+		res = 0;
+		goto exit;
+	}
+
+	if (!buf)
+		goto exit;
+
+	/* fill in parameters */
+	head.op = TEE_FS_WRITE;
+	head.fd = fd;
+	head.len = len;
+
+	res = tee_fs_send_cmd(&head, (void *)buf, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+exit:
+	return res;
+}
+
+static int ree_fs_ftruncate(int fd, tee_fs_off_t length)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+
+	head.op = TEE_FS_TRUNC;
+	head.fd = fd;
+	head.arg = length;
+
+	res = tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
+	if (!res)
+		res = head.res;
+
+	return res;
+}
+
+static int ree_fs_close(int fd)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+
+	/* fill in parameters */
+	head.op = TEE_FS_CLOSE;
+	head.fd = fd;
+
+	res = tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
+	if (!res)
+		res = head.res;
+
+	return res;
+}
+
+static tee_fs_off_t ree_fs_lseek(int fd, tee_fs_off_t offset, int whence)
+{
+	tee_fs_off_t res = -1;
+	struct tee_fs_rpc head = { 0 };
+
+	/* fill in parameters */
+	head.op = TEE_FS_SEEK;
+	head.fd = fd;
+	head.arg = offset;
+	head.flags = whence;
+
+	res = tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
+	if (!res)
+		res = head.res;
+
+	return res;
+}
+
+static int ree_fs_mkdir(const char *path, tee_fs_mode_t mode)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+
+	len = strlen(path) + 1;
+	if (len > REE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_MKDIR;
+	head.flags = mode;
+
+	res = tee_fs_send_cmd(&head, (void *)path, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	return res;
+}
+
+static int ree_fs_rmdir(const char *name)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+
+	len = strlen(name) + 1;
+	if (len > REE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_RMDIR;
+
+	res = tee_fs_send_cmd(&head, (void *)name, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	return res;
+}
+
+static int ree_fs_link(const char *old, const char *new)
+{
+	int res = -1;
+	char *tmp = NULL;
+	struct tee_fs_rpc head = { 0 };
+	size_t len_old = strlen(old) + 1;
+	size_t len_new = strlen(new) + 1;
+	size_t len = len_old + len_new;
+
+	if (len_old > REE_FS_NAME_MAX || len_new > REE_FS_NAME_MAX)
+		goto exit;
+
+	tmp = malloc(len);
+	if (!tmp)
+		goto exit;
+	memcpy(tmp, old, len_old);
+	memcpy(tmp + len_old, new, len_new);
+
+	head.op = TEE_FS_LINK;
+
+	res = tee_fs_send_cmd(&head, tmp, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	free(tmp);
+	return res;
+}
+
+static int ree_fs_unlink(const char *file)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	size_t len = strlen(file) + 1;
+
+	if (len > REE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_UNLINK;
+
+	res = tee_fs_send_cmd(&head, (void *)file, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+exit:
+	return res;
+}
+
+static int get_file_length(int fd, size_t *length)
+{
+	size_t file_len;
+	int res;
+
+	*length = 0;
```

Comment:
Actually this function is just created for increasing code readability, but I'm not sure if the function will be a tool and used in storage subsystem in the future. Adding NULL check would be a good idea here.